bindinfo/cache.go

```diff
@@ -191,3 +207,79 @@ func (b cache) appendNode(newBindRecord *bindRecord, sparser *parser.Parser) err
 	b[hash] = append(b[hash], newNode)
 	return nil
 }
+
+// AddGlobalBind implements GlobalBindAccessor.AddGlobalBind interface.
+func (h *Handle) AddGlobalBind(originSQL, bindSQL, defaultDB, charset, collation string) error {
+	h.lock.Lock()
+	defer h.lock.Unlock()
+
+	ctx := context.TODO()
+	exec, _ := h.ctx.(sqlexec.SQLExecutor)
+	_, err := exec.Execute(ctx, "BEGIN")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			_, err = exec.Execute(ctx, "COMMIT")
+		} else {
+			_, rbErr := exec.Execute(ctx, "ROLLBACK")
+			terror.Log(rbErr)
+		}
+	}()
+
+	sql := fmt.Sprintf("SELECT _tidb_rowid,status FROM mysql.bind_info WHERE original_sql='%s' AND default_db='%s'",
+		originSQL, defaultDB)
+	rs, err := exec.Execute(ctx, sql)
```

Comment:
I'm not sure whether this `err` be caught in `defer`?
Have you tested it?

Reply:
Yes, 
```    package main

import (
	"errors"
	"fmt"
)

func main() {
	xx()
}

func xx(){
	var err error
	defer func() {
		if err != nil {
			fmt.Println("roll back")
		} else {
			fmt.Println("commit")
		}
	}()

	err = nil


	fmt.Println("xxx")

	i,err := yy()

	if err != nil {
		fmt.Println("i=" , i)
		return
	}
}

func yy() (int,error) {

	return 0, errors.New("xxxx")
}
```
consle print roll back